massa-models/src/address.rs

```diff
@@ -14,40 +14,62 @@ use std::str::FromStr;
 /// Derived from a public key
 #[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
 pub struct Address(pub Hash);
-const ADDRESS_STRING_PREFIX: &str = "ADR";
+
+const ADDRESS_PREFIX: char = 'A';
+const ADDRESS_VERSION: u64 = 0;
 
 impl std::fmt::Display for Address {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        if cfg!(feature = "hash-prefix") {
-            write!(f, "{}-{}", ADDRESS_STRING_PREFIX, self.0.to_bs58_check())
-        } else {
-            write!(f, "{}", self.0.to_bs58_check())
-        }
+        // note: might want to allocate the vector with capacity
+        // in order to avoid re-allocation
+        let mut bytes: Vec<u8> = ADDRESS_VERSION.to_varint_bytes();
+        bytes.extend(self.0.to_bytes());
+        write!(
+            f,
+            "{}{}",
+            ADDRESS_PREFIX,
```

Comment:
I think just an A in front of the address will not help people to distinguish when reading fast. Maybe add a dash between the A and the address can help.

Reply:
My concern was that we can have A in the address so people can just consider it as a part of the address. But I see your point